shared/src/main/scala/coop/rchain/shared/AtomicMonadState.scala

```diff
@@ -0,0 +1,15 @@
+package coop.rchain.shared
+
+import cats._
+import cats.implicits._
+import cats.mtl.MonadState
+
+import monix.execution.atomic.Atomic
+
+class AtomicMonadState[F[_], S](state: Atomic[S])(implicit val monad: Monad[F])
+    extends MonadState[F, S] {
+  def get: F[S]                   = state.get.pure[F]
```

Comment:
no! this is effectful!!! `pure` can not be used - pure is only for evaluated values, here you are acitng on atomic reference, you make an effect and then you closed a evaluated value to F.